### PR TITLE
#1758 Added methods for Does.ContainKey and Does.ContainValue, added …

### LINKED ISSUE #1758: No direct inverse for Contains.Key
The `DictionaryContainsKeyConstraint` allows an assertion to be written against the presence of a key in a dictionary:
```
Assert.That(dictionary, Contains.Key("test"));
```
However it's not possible to _directly_ write the inverse - that a dictionary does not contain a key:
```
Assert.That(dictionary, Does.Not.ContainKey("test"));
```
The closest is probably to inspect the dictionary's `Keys` property and assert on that:
```
Assert.That(dictionary.Keys, Does.Not.Contains("test"));
```
Aside from the lack of symmetry, the grammar of this approach is slightly jarring and a dedicated `ContainKey` method would be preferable.

As an aside it's possible to solve this locally in the test code using an extension method:
```
public static class Extensions
{
	public static DictionaryContainsKeyConstraint ContainKey(this ConstraintExpression constraint, string value)
	{
		var keyConstraint = new DictionaryContainsKeyConstraint(value);

		constraint.Append(keyConstraint);

		return keyConstraint;
	}
}
```

### PR DESCRIPTION
**Pull request by JustinRChou**
…units tests.

Added methods for Does.ContainKey and Does.ContainValue.

Added unit tests for new methods. Not sure if they are sufficient.

Added unit test for dictionaries under
CollectionContainsConstraintTests.

Fixes #1758

### CODE REVIEW

File: src/NUnitFramework/tests/Constraints/CollectionContainsConstraintTests.cs [hunk 1]
```diff
@@ -51,6 +51,32 @@ public void CanTestContentsOfObjectList()
             Assert.That(list, new CollectionContainsConstraint(item));
         }
 
+        [Test]
+        public void CanTestContentsOfDictionary()
```
Comment (rprouse) [inline src/NUnitFramework/tests/Constraints/CollectionContainsConstraintTests.cs]: This is a pretty massive test that does many things. I would prefer if it was broken down into smaller tests. It looks like your kvExists123 are good candidates for a `TestCaseSource` and your kvDoesNotExist for a separate test?

**Comment by CharliePoole** [comment]
I agree - good to merge if the test is fixed.

File: src/NUnitFramework/tests/Constraints/CollectionContainsConstraintTests.cs (hunk 1 above)
Comment (JustinRChou) [inline src/NUnitFramework/tests/Constraints/CollectionContainsConstraintTests.cs]: Do i keep the CollectionContainsConstraint test in CollectionContainsConstraintTests and move the Contain asserts to TestCaseSource or move everything to TestCaseSource?

**Comment by rprouse** [comment]
You can leave the tests here. I just suggested TestCaseSource as a way of writing the tests to reduce duplication.

**Comment by JustinRChou** [comment]
I see now. Rewriting it using the testsource attributes then.

**Comment by JustinRChou** [comment]
Is it an issue when Does.Contain refers to the CollectionContainsConstraint but Does.Not.Contain refers to ContainsConstraint?

It was causing me some confusion.

**Comment by CharliePoole** [comment]
@JustinRChou You appear to have added a commit that's intended for issue #1808. You should use different local branches and create different PRs for resolving each issue, unless two issues have some close relationship to one another. That makes it faster to finish reviewing each thing and get it merged. Adding new stuff makes it a moving target for us.

As far as I can tell, the last commit that relates to issue #1758 is the one with the description "Reorganize the unit test" Is that correct? If you need help resetting the branch to that commit while preserving your new work on a new local branch, let me know. Meanwhile, this branch and PR should wait for @rprouse to review the change he requested.

**Comment by CharliePoole** [comment]
Looks good now. Just waiting for @rprouse to re-review.

**Comment by CharliePoole** [comment]
@rprouse This can't be merged until you re-review, according to the new review system. That's one of the things I dislike about how it works. [I realize I can override and merge it, but that's not playing by the rules.]

**Comment by rprouse** [comment]
Interesting point on the review system. I didn't realize another team member can't just switch to Approved. We'll need to think on that...

**Comment by CharliePoole** [comment]
Most likely, you or I as owners can do anything. But we shouldn't rely on
that.

**Pull request merged**